odoo/models.py

```diff
@@ -1761,15 +1763,22 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
         :return: list of pairs ``(id, display_name)`` for all matching records.
         """
         ids = self._name_search(name, args, operator, limit=limit, order=self._order)
-
+        extra_fields = extra_fields or []
+        extra_fields = [field for field in extra_fields if field in self._fields and field in self._name_search_allowed_extra_fields]
         if isinstance(ids, Query):
-            records = self._fetch_query(ids, self._determine_fields_to_fetch(['display_name']))
+            records = self._fetch_query(ids, self._determine_fields_to_fetch(['display_name', *extra_fields]))
         else:
             # Some override of `_name_search` return list of ids.
             records = self.browse(ids)
-            records.fetch(['display_name'])
-
-        return [(record.id, record.display_name) for record in records.sudo()]
+            records.fetch(['display_name', *extra_fields])
+        res = []
+        for record in records.sudo():
+            extra = ({field: record[field] for field in extra_fields})
+            if extra:
+                res.append((record.id, record.display_name, extra))
+            else:
+                res.append((record.id, record.display_name))
```

Comment:
So this returns triples or pairs depending on whether the extra fields are allowed or not, right?  In other words, the type of the output depends on parameter `extra_fields` and other stuff you don't have access to. 
 Prepare for typing errors client-side!  Unless I am mistaken, this case (extra fields that are not allowed) does not seem covered by tests...